### PR TITLE
Add go to top button

### PR DESCRIPTION
**Pull request by rahat2134**
Fix issue : #47 
Screenshot 2024-01-24 at 11 02 59 AM

### CODE REVIEW

**Comment by rahat2134** [comment]
@JiyaGupta-cs please check the pr.

**Comment by JiyaGupta-cs** [comment]
@rahat2134 It works only when you scroll in the "What's happening section" but not in the Posts section

**Comment by rahat2134** [comment]
> @rahat2134 It works only when you scroll in the "What's happening section" but not in the Posts section

@JiyaGupta-cs  i tried too much things to add gototop button on scroll bar but its not working, can you help me. In which folder i need to add component of button, I tried most of website but the button remain on the screen and not in scroll/post section.

**Comment by varshith257** [comment]
@rahat2134  It can be resolved by using

> > @rahat2134 It works only when you scroll in the "What's happening section" but not in the Posts section
> 
> @JiyaGupta-cs i tried too much things to add gototop button on scroll bar but its not working, can you help me. In which folder i need to add component of button, I tried most of website but the button remain on the screen and not in scroll/post section.

@rahat2134  This can be resolved by adding a second `useEffect `block  with use  of `EventListener` to the` window` to listen for a scroll  for Posts container

For your reference:  https://stackabuse.com/how-to-scroll-to-top-in-react-with-a-button-component/

**Comment by JiyaGupta-cs** [comment]
@rahat2134 Will you work on it? Or @varshith257 Do you wish to take up the issue?

**Comment by rahat2134** [comment]
@varshith257 go for it.

**Comment by varshith257** [comment]
> @rahat2134 Will you work on it? Or @varshith257 Do you wish to take up the issue?

@JiyaGupta-cs Let them work. :) I will provide any guidance if required. Let's ensure their efforts are not in vain and guide them to build progress

**Comment by varshith257** [comment]
> @varshith257 go for it.

@rahat2134  Get Spirit it man! You can solve it. You have made efforts to solve it with bit changes you can successfully solve it

**Review by varshith257 — CHANGES_REQUESTED** [review]
move the whole code in return function into `post-container`.  After changes It could look like : 

```
const Post = ({ displayName, userName, verified, text, image, avatar, location }) => {
  return (
    <div id="posts-container" className='post-container'>
      <div className='post'>
        <div className="post_avatar">
          <img src={avatar} alt="" />
        </div>
        <GoToTopButton />

        <div className="post_body">
          .........................
```

File: src/components/Button/GoToTopButton.jsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    }
+  };
+
+  useEffect(() => {
+    window.addEventListener('scroll', handleScroll);
+    return () => {
+      window.removeEventListener('scroll', handleScroll);
+    };
+  }, []);
+
```
[the remark this answers is not in the document]
Comment (rahat2134) [reply src/components/Button/GoToTopButton.jsx:37]: Bandwidth issue

**Comment by JiyaGupta-cs** [comment]
@varshith257 ok

**Comment by JiyaGupta-cs** [comment]
@varshith257 Fix the issue

**Comment by JiyaGupta-cs** [comment]
@varshith257 Comment on the issue and I will assign you at the issue

**Comment by varshith257** [comment]
> @varshith257 Comment on the issue and I will assign you at the issue

@JiyaGupta-cs Mybad! Sorry for the late reply. I am not active here due to exams. Someone has fixed it  let them work

**Pull request closed without merging**